### PR TITLE
RUST-1108 Add a manual section on client best practices

### PR DESCRIPTION
**Pull request by abr-egn**
RUST-1108

This covers client lifetime, pooling, runtime pitfalls, and encouraged parallelism.

### CODE REVIEW

File: manual/src/performance.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    // Do something with the client
+    Ok(())
+}
+```
+
+This is especially noticeable when using a framework that provides connection pooling; because `Client` does its own pooling internally, attempting to maintain a pool of `Client`s will (somewhat counter-intuitively) result in worse performance than using a single one.
+
+### Runtime
+
+A `Client` is implicitly bound to the instance of the `tokio` or `async-src` runtime in which it was created.  Attempting to execute operations on a different runtime instance will cause incorrect behavior and unpredictable failures.  This is easy to accidentally invoke when testing, as the `tokio::test` or `async_std::test` helper macros create a new runtime for each test.
```
Comment (kmahar) [inline manual/src/performance.md:32]: async-src -> async-std?

**Review by kmahar — APPROVED** [review]
LGTM mod one possible typo

File: manual/src/performance.md (hunk 1 above)
Comment (abr-egn) [reply manual/src/performance.md]: Derp, finger macro failure.  Fixed.

File: manual/src/performance.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Performance
+
+## `Client` Best Practices
+
+The [`Client`](https://docs.rs/mongodb/latest/mongodb/struct.Client.html) handles many aspects of databse connection behind the scenes that can require manual management for other database drivers; it discovers server topology, monitors it for any changes, and maintains an internal connection pool.  This has implications for how a `Client` should be used for best performance.
```
Comment (isabelatkinson) [inline manual/src/performance.md:5]:
```suggestion
The [`Client`](https://docs.rs/mongodb/latest/mongodb/struct.Client.html) handles many aspects of database connection behind the scenes that can require manual management for other database drivers; it discovers server topology, monitors it for any changes, and maintains an internal connection pool.  This has implications for how a `Client` should be used for best performance.
```

**Review by isabelatkinson — APPROVED** [review]
lgtm! just one typo fix

File: manual/src/performance.md (hunk 2 above)
Comment (abr-egn) [reply manual/src/performance.md]: Good catch, fixed.

File: manual/src/performance.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+use once_cell::sync::Lazy;
+
+static CLIENT_RUNTIME: Lazy<(Client, Runtime)> = Lazy::new(|| {
+    let rt = Runtime::new().unwrap();
+    let client = rt.block_on(async {
+        Client::with_uri_str("mongodb://example.com").await.unwrap()
+    });
+    (client, rt)
+});
+
```
Comment (patrickfreed) [inline manual/src/performance.md:73]: can we add `#[test]` here to emphasize that `tokio::test` shouldn't be used with this scheme.

**Review by patrickfreed — COMMENTED** [review]
one small suggestion but otherwise LGTM

**Review by patrickfreed — APPROVED** [review]

**Pull request merged**